src/main/java/net/masterthought/cucumber/generators/AbstractPage.java

```diff
@@ -64,58 +34,12 @@ public void generatePage() {
      * @return HTML file for the report
      */
     public abstract String getWebPage();
+    
+    /**
+     * Populates the given context to merge it with the template
+     * 
+     * @param context
+     */
+    public abstract void preparePageContext(VelocityContext context, Configuration configuration, ReportResult reportResult);
 
-    protected abstract void prepareReport();
-
-    private void generateReport() {
```

Comment:
can you explain this refactor?

Reply:
The idea behind the refactor is to apply the single responsibility principle. An AbstractPage is the link between the velocity template and the necessary data to render the given template. 

The PageGenerator object is the one who initialize the VelocityEngine and using an AbstractPage object render the template with the given data writing it to disk.